app/custom_domain_validation.py

```diff
@@ -0,0 +1,33 @@
+from app.db import Session
+from app.dns_utils import get_cname_record
+from app.models import CustomDomain
+
+
+class CustomDomainValidation:
```

Comment:
I wonder if we can move this dkim checking to `dns_utils`? Maybe we don't need the whole object `CustomDomain ` and just the `domain` (`str`)?

Reply:
Something like 

```
DKIM_SELECTOR = ("dkim", "dkim02", "dkim03")
def validate_dkim_records(domain: str) -> bool:
    """
    Return true if dkim records are properly set for this domain.
    """
    invalid_records = {}
    prefix_and_records = {
            (f"{key}._domainkey", f"{key}._domainkey.{EMAIL_DOMAIN}")
            for key in DKIM_SELECTOR
        }
    for prefix, expected_record in prefix_and_records:
        custom_record = f"{prefix}.{domain}"
        dkim_record = get_cname_record(custom_record)
        if dkim_record != expected_record:
            invalid_records[custom_record] = dkim_record or "empty"
    return len(invalid_records) == 0
```

Reply:
I want to eventually move all domain validations out of the controller logic to be able to reuse it in other parts (like via api) if needed. So this is just the first one that will go into the Validator. But I'll add the refactor in another MR.

Reply:
Ah I think we don't need OOP to have code reusuable right? Currently `dns_utils` doesn't have much dependency and can be used everywhere.

Reply:
It moves us closer to removing global vars. If we have a function in a module we still need to have something global to that module and can't avoid having global variables.